### PR TITLE
Feature/rtofs in situ

### PR DESCRIPTION
**Pull request by givelberg**
Created 2 converters with corresponding tests, for temperature and salinity.
2 different files are created.
For temperature the file includes
tmp, tmp_err, lon, lat, depth
and metadata sal_typ and flg
Similarly, for salinity..

### CODE REVIEW

File: utils/test/CMakeLists.txt [hunk 1]
```diff
@@ -63,6 +65,15 @@ ecbuild_add_test( TARGET  test_gdasapp_util_ghrsst2ioda
                   LIBS    gdas-utils
                   WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/obsproc)
 
+#TODO:
+# add binary files for the test
```
Comment (guillaumevernieres) [inline utils/test/CMakeLists.txt]: Lets stage the binary files on Hera and Orion so we can switch that test on

File: utils/test/testinput/gdas_rtofsinsitu.yaml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+provider: RTOFS
+window begin: 2021-03-24T15:00:00Z
+window end: 2021-03-24T21:00:00Z
+output file: rtofsinsitu_2024032600.profile.nc4
+input files:
```
Comment (guillaumevernieres) [inline utils/test/testinput/gdas_rtofsinsitu.yaml]: we need to be able to read multiple files

File: utils/test/testinput/gdas_rtofsinsitu.yaml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+provider: RTOFS
+window begin: 2021-03-24T15:00:00Z
+window end: 2021-03-24T21:00:00Z
+output file: rtofsinsitu_2024032600.profile.nc4
+input files:
+- rtofsinsitu_2024032600.profile
+# - 2024032600.profile
+# - 2024032600.sfc
+
+# test:
```
Comment (guillaumevernieres) [inline utils/test/testinput/gdas_rtofsinsitu.yaml]: test reference needs to be used

**Review by guillaumevernieres — COMMENTED** [review]
Thanks for doing this @givelberg . A few minor comments related to the ctest. We'll figure out where to keep these RTOFS binaries next week.

**Comment by guillaumevernieres** [comment]
Looks like you forgot to add the yaml files for the tests @givelberg :
```
 6/48 Test #1493: test_gdasapp_util_rtofstmp ............................Subprocess aborted***Exception:   0.82 sec                                                                                                        
Exception: Cannot open ../testinput/gdas_rtofstmp.yaml  (No such file or directory)                                                                                                                                        
terminate called after throwing an instance of 'eckit::CantOpenFile'                                                                                                                                                       
  what():  Cannot open ../testinput/gdas_rtofstmp.yaml  (No such file or directory)                                                                                                                                        
                                                                                                                                                                                                                           
      Start 1494: test_gdasapp_util_rtofssal                                                                                                                                                                               
 7/48 Test #1494: test_gdasapp_util_rtofssal ............................Subprocess aborted***Exception:   0.60 sec                                                                                                        
Exception: Cannot open ../testinput/gdas_rtofssal.yaml  (No such file or directory)                                                                                                                                        
terminate called after throwing an instance of 'eckit::CantOpenFile'                                                                                                                                                       
  what():  Cannot open ../testinput/gdas_rtofssal.yaml  (No such file or directory)             
```

File: utils/test/CMakeLists.txt [hunk 4]
```diff
@@ -65,15 +70,59 @@ ecbuild_add_test( TARGET  test_gdasapp_util_ghrsst2ioda
...
+				  ARGS    "../testinput/gdas_rtofssal.yaml"
+                  LIBS    gdas-utils
+                  WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/obsproc)
+	else()
+		message("Error: RTOFS input files not found; no test generated.")
+	endif()
+endif()
+
+
+<<<<<<< HEAD
```
Comment (guillaumevernieres) [inline utils/test/CMakeLists.txt]: Ha! I'm a terrible reviewer!!!
The build failed because of this @givelberg

**Review by guillaumevernieres — CHANGES_REQUESTED** [review]
See comment. I didn't see any other ```HEAD``` or residuals from your last merge, but you might want to double check.
Also, you need to merge develop into your branch again.

**Review by guillaumevernieres — APPROVED** [review]
Thanks @givelberg !

**Comment by RussTreadon-NOAA** [comment]
This PR changed the hash for the following repos in `sorc/`:  `fv3`, `gsibec`, `gsw`, `land-imsproc`, `land-jediincr`.   The modified hashes in this PR are newer than what was previously in GDASApp `sorc/` except for `gsibec`.

This PR changed gsibec to `[gsibec @ e6e1944](https://github.com/GEOS-ESM/GSIbec/tree/e6e1944e1cb8f339b4eb79acffb5470bb7723b42).  This commit is dated June 26, 2203.

Before this PR was merged into `develop`, we were using [gsibec @ c8ac58d](https://github.com/GEOS-ESM/GSIbec/tree/c8ac58d9b43eb8f890e565d12c88f1b0579c9ccd).  This commit is dated March 8, 2024.  This gsibec hash is [tags/1.2.1](https://github.com/GEOS-ESM/GSIbec/releases/tag/1.2.1).

g-w C96C48_ufs_hybatmDA CI using the current head of GDASApp ([7f4c2fc](https://github.com/NOAA-EMC/GDASApp/commit/7f4c2fc15bcf6c4bdc212b3d64f107fd92fda2ad)) failed when running the gfs and gdas atmanlvar.  Both job aborted with the error message
```
 0: Exception: gsi static covariance does not exist in saber::SaberCentralBlockFactory
 0: Exception: oops::Variational<FV3JEDI, UFO and IODA observations> terminating...
 0: Exception: level: 0
 0: gsi static covariance does not exist in saber::SaberCentralBlockFactory
 0: OOPS Ending   2024-06-11 16:39:22 (UTC+0000)
```
This reason for this failure is that the `gsibec` hash GDASApp currently points at is out of date with other repository snapshots in `sorc/`.  As a test `gsibec` was updated to gsibec `tags/1.2.1` and GDASApp rebuilt.  Reruns of the g-w CI gfs and gdas atmanlvar jobs were successful.

**We need to revert `sorc/gsibec` back to gsibec [`tags/1.2.1`](https://github.com/GEOS-ESM/GSIbec/releases/tag/1.2.1) (hash [c8ac58d](https://github.com/GEOS-ESM/GSIbec/commit/c8ac58d9b43eb8f890e565d12c88f1b0579c9ccd))**

**Comment by guillaumevernieres** [comment]
Ouch, sorry about this @RussTreadon-NOAA . Not quite sure how I missed this but I did ...

**Pull request merged**